app/models/ability.rb

```diff
@@ -29,4 +31,12 @@ def initialize(user)
       can :read, ActiveAdmin
     end
   end
+
+  private
+
+  MINIMUM_DAYS_TO_CANCEL = 7
+
+  def within_cancel_range?(vacation)
+    vacation.approved? && (vacation.start_date.days_ago(MINIMUM_DAYS_TO_CANCEL) >= Date.today)
```

Comment:
From the use case perspective, I'm thinking if we really need to check if the vacation was approved or not, maybe if it's within the cancel range that's enough and we can ditch the `vacation.approved?` check. 
cc @adbatista @JonasTeixeira42

Reply:
I think this is necessary to keep it because if we remove the status, the cancel icon will display for all vacations that are not in `pending` status

Reply:
the user can cancel in these cases
 1. vacation pending
 2. vacation approved and start date > 7 days from today

the user can not cancel in these cases
 1. Vacation canceled
 2. Vacation Rejected
 3. Vacation approved and the start date <= 7 days from today